### PR TITLE
Creates Rocket and RocketController class. Fulfills #14

### PR DESCRIPTION
**Pull request by karandesai-96**
Magnus can now fire a rocket which follows the finger position for sometime and finally exhausts.

### CODE REVIEW

**Comment by karandesai-96** [comment]
@hareshkh I think we are good to go here. Please review it.

File: core/src/com/sdsmdg/kd/gameplay/objects/Rocket.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        // Setting the radius such that the rocket is (1/18)th the size of screen.
+        this.radius = (int) Math.sqrt(Main.screenArea / (18 * MathUtils.PI));
+
+        // The rocket should be out of screen when the game starts, hence its
+        // coordinates are set a little outside the screen.
+        this.x = Main.screen.x + (4 * this.radius);
+        this.y = Main.screen.y + (4 * this.radius);
+
+        // Velocity of Rocket can be anything between 10 and 24 units.
+        this.velocity = random.nextInt(15) + 10;
```
Comment (hareshkh) [inline core/src/com/sdsmdg/kd/gameplay/objects/Rocket.java]: random is instantiated after its usage and will cause Null Pointer exception.

File: core/src/com/sdsmdg/kd/gameworld/GameWorld.java [hunk 2]
```diff
@@ -74,14 +86,14 @@ public void update(float delta) {
...
-        currentWeapon = 1;
-
-        /* when Bullets are selected to be fired, initial point and direction
-         * are set using initBullets method.
-         *
-         * @param magnus For using the coordinates of its center.
-         */
-
-        bullet.initBullets(magnus);
+        currentWeapon = random.nextInt(2);
```
Comment (hareshkh) [inline core/src/com/sdsmdg/kd/gameworld/GameWorld.java]: random used here is declared and then used but not instantiated and must be assigned an instance before usage.

Comment (karandesai-96) [inline core/src/com/sdsmdg/kd/gameworld/GameWorld.java]: Now I am lazy enough to look above as I can only see partial diff, but you didn't find `= new RandomXS128` written in constructor ? :thought_balloon:

File: core/src/com/sdsmdg/kd/gameplay/objects/Rocket.java (hunk 1 above)
Comment (karandesai-96) [inline core/src/com/sdsmdg/kd/gameplay/objects/Rocket.java]: Nice
- [X] Done.

File: core/src/com/sdsmdg/kd/gameworld/GameWorld.java (hunk 2 above)
Comment (hareshkh) [inline core/src/com/sdsmdg/kd/gameworld/GameWorld.java]: For the variable currentWeapon to be equal to 2 which is necessary for selection of Rocket as a weapon, the parameter here must be 3.

Comment (karandesai-96) [inline core/src/com/sdsmdg/kd/gameworld/GameWorld.java]: Right, this needs to be done.

File: core/src/com/sdsmdg/kd/gameworld/GameRenderer.java [hunk 3]
```diff
@@ -33,19 +33,19 @@ public void render() {
...
 
-        //Tells the shape renderer to finish rendering. This must be done every time.
-        shapeRenderer.end();
+        //Set up (Red,Green,Blue,Alpha)
+        shapeRenderer.setColor(0 / 255.0f, 0 / 255.0f, 255 / 255.0f, 1);
 
-        //Begin drawing filled shape
-        shapeRenderer.begin(ShapeType.Filled);
+        //Draws circle from GameWorld for Bullet.
+        shapeRenderer.circle(gameWorld.bullet.x, gameWorld.bullet.y, gameWorld.bullet.radius);
```
Comment (karandesai-96) [inline core/src/com/sdsmdg/kd/gameworld/GameRenderer.java]: Bullet would be rendered one above another, replace `bullet` with `rocket` here.

**Comment by hareshkh** [comment]
Review this @karandesai-96. I think with the latest commit, the objective of the PR is achieved.

**Comment by karandesai-96** [comment]
Everything looks good. Merging

**Pull request merged**